aws/run-on-ec2.py

```diff
@@ -115,6 +116,14 @@ def getButterflyNetworkCommands(s3Manager, instanceIps, instanceIds):
     return AwsConfig.MPC_CONFIG.COMMAND, setupCommands
 
 
+q = Queue()
```

Comment:
I don't think having a global variable is a good idea, unless it is a constant. Can you please move it within one of the functions and then pass it around?